File: programmers/level2/타겟넘버.py
def solution(numbers, target):
    if len(numbers) == 0 and target==0: return 1
    elif len(numbers)==0: return 0
    return solution(numbers[1:], target-numbers[0]) + solution(numbers[1:], target+numbers[0]) #재귀함수로 앞에서부터 타겟넘버에서 더하거나 뺀다

File: programmers/level2/타겟 넘버.java
class Solution {
    private int[] numbers;
    private int target;
    private int answer = 0;
    public int solution(int[] numbers, int target) {
        this.numbers = numbers;
        this.target = target;
        dfs(0, numbers[0]);
        dfs(0, -numbers[0]);
        return answer;
    }
    
    public void dfs(int idx, int sum) {
        if(idx == numbers.length - 1 && sum == target) {
            answer += 1;
            return;
        }
        if(idx == numbers.length - 1) return;
        dfs(idx+1, sum+numbers[idx+1]);
        dfs(idx+1, sum-numbers[idx+1]);
    }
}